internal/charm/repository/charmhub.go

```diff
@@ -337,17 +337,12 @@ func (c *CharmHubRepository) resolveWithPreferredChannel(ctx context.Context, ch
 
 // validateOrigin, validate the origin and maybe fix as follows:
 //
-//	Platform must have an architecture.
 //	Platform can have both an empty Channel AND os.
 //	Platform must have channel if os defined.
 //	Platform must have os if channel defined.
 func (c *CharmHubRepository) validateOrigin(origin corecharm.Origin) (corecharm.Origin, error) {
 	p := origin.Platform
 
-	if p.Architecture == "" {
-		return corecharm.Origin{}, errors.BadRequestf("origin.Platform requires an Architecture")
-	}
```

Comment:
`Architecture` is no longer required, we can handle empty arch by defaulting

Reply:
https://github.com/juju/juju/blob/1922a1c03cc600d9f7f95a8deb1d032bbcb8a242/internal/charm/repository/charmhub.go#L179-L194